### PR TITLE
get current color

### PR DESCRIPTION
**Pull request by nobody5050**

### CODE REVIEW

File: connector_x/src/main.cpp [hunk 1]
```diff
@@ -119,20 +116,20 @@ Adafruit_NeoPixel *getPixels(uint8_t port)
...
+        return pixels[0];
     }
 
-    return pixels1;
+    return pixels[1];
 }
 
 PatternRunner *getPatternRunner(uint8_t port)
 {
     if (port == 0)
```
Comment (having11) [inline connector_x/src/main.cpp]: Why not check if the port is within the limit and then do `patternRunners[port]`?

File: connector_x/src/main.cpp [hunk 2]
```diff
@@ -164,8 +161,8 @@ void loop()
         case CommandType::On:
         {
             // Go back to running the current color and pattern
-            patternRunner0->reset();
-            patternRunner1->reset();
+            patternRunner[0]->reset();
```
Comment (having11) [inline connector_x/src/main.cpp]: For-loop here

File: connector_x/src/main.cpp [hunk 3]
```diff
@@ -265,8 +273,8 @@ void loop()
 
     if (systemOn)
     {
-        patternRunner0->update();
-        patternRunner1->update();
+        patternRunner[0]->update();
```
Comment (having11) [inline connector_x/src/main.cpp]: For-loop here

File: connector_x/src/main.cpp [hunk 4]
```diff
@@ -240,6 +237,17 @@ void loop()
             break;
         }
 
+        case CommandType::GetColor:
```
Comment (having11) [inline connector_x/src/main.cpp]: Why is a read operation under the write operation switch block?

File: connector_x/src/main.cpp [hunk 5]
```diff
@@ -240,6 +237,17 @@ void loop()
...
+        {
+            auto runner = getPatternRunner(ledPort);
+
+            mutex_enter_blocking(&i2cCommandMtx);
```
Comment (having11) [inline connector_x/src/main.cpp]: Why is a mutex necessary here when we're using a single core?

File: connector_x/src/main.cpp [hunk 6]
```diff
@@ -347,6 +347,15 @@ void requestEvent()
         break;
     }
 
+    case CommandType::GetColor:
+    {
+        auto runner = getPatternRunner(ledPort);
+
+        uint32_t msg = runner->getCurrentColor();
+
+        res.responseData.responseReadColor.msg = msg;
```
Comment (having11) [inline connector_x/src/main.cpp:355]: I don't see a handler in `centralRespond()`

File: connector_x/src/main.cpp [hunk 7]
```diff
@@ -347,6 +347,22 @@ void requestEvent()
...
+    case CommandType::GetColor:
+    {
+        auto runner = getPatternRunner(ledPort);
+        uint32_t msg = runner->getCurrentColor();
+
+        res.responseData.responseReadColor.msg = msg;
+        break;
+    }
+
+    case CommandType::GetPort:
```
Comment (having11) [inline connector_x/src/main.cpp:359]: Need a handler for this too

File: connector_x/include/Commands.h [hunk 8]
```diff
@@ -159,12 +163,24 @@ struct ResponseReadConfiguration
     Configuration config;
 };
 
+struct ResponseReadColor
+{
+    uint32_t color;
+}
```
Comment (having11) [inline connector_x/include/Commands.h]: semicolon

File: connector_x/include/Commands.h [hunk 9]
```diff
@@ -159,12 +163,24 @@ struct ResponseReadConfiguration
...
+
+struct ResponseReadPort
+{
+    uint8_t port;
+}
```
Comment (having11) [inline connector_x/include/Commands.h]: Semicolon

File: connector_x/include/Commands.h [hunk 10]
```diff
@@ -31,7 +31,11 @@ enum class CommandType
     // W
     RadioSend = 12,
     // R
-    RadioGetLatestReceived = 13
+    RadioGetLatestReceived = 13,
+    // R
+    GetColor = 14,
+    // R
+    GetPort = 15
```
Comment (having11) [inline connector_x/include/Commands.h:38]: Need to create new, empty `CommandGetColor` and `CommandGetPort` structs

Comment (having11) [reply connector_x/include/Commands.h:38]: Also need to add handlers to CommandParser.h

**Review by having11 — CHANGES_REQUESTED** [review]

File: connector_x/include/Commands.h (hunk 10 above)
Comment (nobody5050) [reply connector_x/include/Commands.h:38]: > Also need to add handlers to CommandParser.h

Resolved conversation for original comment, unsure how to do the second comment

Comment (having11) [reply connector_x/include/Commands.h:38]: Just add new cases within CommandParser.h > `parseCommand` to handle setting the two command structs.

Comment (having11) [reply connector_x/include/Commands.h:38]: > Just add new cases within CommandParser.h > `parseCommand` to handle setting the two command structs.

Still need this

File: connector_x/src/main.cpp [hunk 11]
```diff
@@ -347,6 +347,22 @@ void requestEvent()
...
+    {
+        uint8_t msg = ledPort;
+        res.responseData.responseReadPort.msg = msg;
```
Comment (having11) [inline connector_x/src/main.cpp]: nit: assign `ledPort` directly to `res.responseData.responseReadPort.msg`

Comment (nobody5050) [reply connector_x/src/main.cpp]: fixed

**Review by having11 — APPROVED** [review]
good

**Pull request merged**